src/helper/remote_config/client.hpp

```diff
@@ -0,0 +1,71 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog
+// (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "http_api.hpp"
+#include "product.hpp"
+#include "protocol/client.hpp"
+#include "protocol/tuf/get_configs_response.hpp"
+
+namespace dds::remote_config {
+
+struct config_path {
+public:
+    config_path() = default;
+    std::string get_id() { return _id; };
+    std::string get_product() { return _product; };
+    void set_id(std::string &id) { _id = id; };
+    void set_product(std::string &product) { _product = product; };
+
+private:
+    std::string _product;
+    std::string _id;
+};
+
+protocol::remote_config_result config_path_from_path(
+    const std::string &path, config_path &cp);
+
+class client {
+public:
+    // NOLINTNEXTLINE(bugprone-easily-swappable-parameters)
+    client(http_api *arg_api, std::string &id, std::string &runtime_id,
+        // NOLINTNEXTLINE(bugprone-easily-swappable-parameters)
+        std::string &tracer_version, std::string &service, std::string &env,
+        std::string &app_version, std::vector<product> &products)
+        : _api(arg_api), _id(id), _runtime_id(runtime_id),
+          _tracer_version(tracer_version), _service(service), _env(env),
+          _app_version(app_version), _targets_version(0)
+    {
+        for (product &p : products) {
+            _products.insert(std::pair<std::string, remote_config::product>(
+                p.get_name(), p));
+        }
+    };
+
+    protocol::remote_config_result poll();
+
+private:
+    protocol::get_configs_request generate_request();
+    protocol::remote_config_result process_response(
+        protocol::get_configs_response response);
+
+    http_api *_api;
```

Comment:
You completely reversed the convention, members don't start with an underscore, they end with one:

```cpp
http_api *api_;
````

Reply:
Done in all the places where I fond this pattern